src/lib/transition_handler/catchup_monitor.ml

```diff
@@ -1,51 +1,119 @@
-        , drop_head buffered
-        , unit )
+        , synchronous
+        , unit Deferred.t )
         Writer.t
     ; timeouts:
-        (State_hash.t, unit Time.Timeout.t) List.Assoc.t State_hash.Table.t }
-
-  let create ~catchup_job_writer =
-    {catchup_job_writer; timeouts= State_hash.Table.create ()}
+        ( (External_transition.Verified.t, State_hash.t) With_hash.t
+        , unit Time.Timeout.t )
+        List.Assoc.t
+        State_hash.Table.t
+    ; breadcrumb_builder_supervisor:
+        (External_transition.Verified.t, State_hash.t) With_hash.t Rose_tree.t
+        list
+        Capped_supervisor.t }
 
-  let watch t ~logger ~time_controller ~timeout_duration ~transition =
+  let create ~logger ~frontier ~time_controller ~catchup_job_writer
+      ~catchup_breadcrumbs_writer =
     let logger = Logger.child logger "catchup_monitor" in
+    let timeouts = State_hash.Table.create () in
+    let breadcrumb_builder_supervisor =
+      Capped_supervisor.create ~job_capacity:5 (fun transition_branches ->
+          let%bind breadcrumbs =
+            Deferred.create (fun ivar ->
+                Ivar.fill ivar
+                  (List.map transition_branches ~f:(fun branch ->
+                       let (Rose_tree.T (branch_base, _)) = branch in
+                       let branch_parent_hash =
+                         With_hash.data branch_base
+                         |> External_transition.Verified.protocol_state
+                         |> Protocol_state.previous_state_hash
+                       in
+                       let branch_parent =
+                         Transition_frontier.find_exn frontier
+                           branch_parent_hash
+                       in
+                       Rose_tree.fold_map branch ~init:branch_parent
+                         ~f:(fun parent transition_with_hash ->
+                           let breadcrumb =
+                             Transition_frontier.Breadcrumb.build ~logger
+                               ~parent ~transition_with_hash
+                             |> Or_error.ok_exn
+                           in
+                           breadcrumb ) )) )
+          in
+          Writer.write catchup_breadcrumbs_writer breadcrumbs )
+    in
+    { logger
+    ; time_controller
+    ; catchup_job_writer
+    ; timeouts
+    ; breadcrumb_builder_supervisor }
+
+  let watch t ~timeout_duration ~transition =
     let hash = With_hash.hash transition in
     let parent_hash =
       With_hash.data transition |> External_transition.Verified.protocol_state
       |> Protocol_state.previous_state_hash
     in
     let make_timeout () =
-      Time.Timeout.create time_controller timeout_duration ~f:(fun _ ->
-          Writer.write t.catchup_job_writer transition )
+      Time.Timeout.create t.time_controller timeout_duration ~f:(fun _ ->
+          don't_wait_for (Writer.write t.catchup_job_writer transition) )
```

Comment:
argue why this is a safe `don't_wait_for` in a comment. I believe it, but I didn't immediately believe it